py3status/modules/mpd_status.py

```diff
@@ -213,7 +213,8 @@ def attr_getter(attr):
             text = "Failed to connect to mpd!"
         except CommandError:
             text = "Failed to authenticate to mpd!"
-            c.disconnect()
+
+        c.disconnect()
```

Comment:
Unfortunately you're right and there is no way to check whether we're connected or not with the mpd module.

Fixed.